utils_test.go

```diff
@@ -1,9 +1,65 @@
 package skynet
 
 import (
+	"net/url"
 	"testing"
 )
 
+const (
+	portalURL = DefaultPortalURL
+)
+
+// TestMakeURL tests making URLs.
+func TestMakeURL(t *testing.T) {
```

Comment:
nit: might be worth making these test cases be the actual API endpoints. Could then be extended every time we add an endpoint.  

The benefit I see is that then we also are explicitly testing the formation of all the urls for the SDK.

Reply:
All the endpoints should be tested by the mock tests, which try to match requests to the expected full URLs. That should be enough to ensure that `makeURL` is working for all the default endpoints. My main concern and reason for this unit test is that people can pass in their own custom endpoint paths, for that I think we just need to test the possible edge cases concerning slashes